pennylane_lightning/core/src/observables/Observables.hpp

```diff
@@ -414,4 +414,108 @@ class HamiltonianBase : public Observable<StateVectorT> {
     }
 };
 
+/**
+ * @brief Sparse representation of HamiltonianGPU<T>
+ *
+ * @tparam T Floating-point precision.
+ */
+template <class StateVectorT>
+class SparseHamiltonianBase : public Observable<StateVectorT> {
+  public:
+    using PrecisionT = typename StateVectorT::PrecisionT;
+    using ComplexT = typename StateVectorT::ComplexT;
+    // cuSparse required index type
+    using IdxT =
+        typename std::conditional<std::is_same<PrecisionT, float>::value,
+                                  int32_t, int64_t>::type;
+
+  protected:
+    std::vector<std::complex<PrecisionT>> data_;
```

Comment:
Do we need to abstract away the container holding the data? to make it GPU-compatible? How does LK work with this?Or is this later inherited from and overridden?

Reply:
Let's remove it to make all backends aligned.